### PR TITLE
Fix update client status

### PR DESCRIPTION
**Pull request by SYangster**
Check workflow id when handling event before calling `_update_client_status()` for the case of multiple workflows

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by holgerroth — DISMISSED** [review]
Looks good to me.

File: nvflare/app_common/ccwf/server_ctl.py [hunk 1]
```diff
@@ -345,7 +345,9 @@ def control_flow(self, abort_signal: Signal, fl_ctx: FLContext):
 
     def handle_event(self, event_type: str, fl_ctx: FLContext):
         if event_type == EventType.BEFORE_PROCESS_TASK_REQUEST:
-            self._update_client_status(fl_ctx)
+            peer_ctx = fl_ctx.get_peer_context()
```
Comment (yanchengnv) [inline nvflare/app_common/ccwf/server_ctl.py:348]: Why is this change necessary? self._update_client_status already checked these conditions.

**Review by yanchengnv — CHANGES_REQUESTED** [review]
Don't see why this change is needed. Did I miss something?

File: nvflare/app_common/ccwf/server_ctl.py (hunk 1 above)
Comment (SYangster) [reply nvflare/app_common/ccwf/server_ctl.py:348]: When there are multiple workflows (e.g swarm and cse), they will both handle this event. 

During the swarm workflow, the cse workflow will also call `_update_client_status()` and at `if client_name not in self.client_statuses:` since self.client_statuses is not set yet, it will continuously log an error `self.log_error(fl_ctx, f"received result from unknown client {client_name}!")`.

Some options are to change it from log_error or check here to see if self.client_statuses is set, but I was thinking to check beforehand to see if the current workflow matches to avoid calling `_update_client_status()` unnecessarily.

Comment (yanchengnv) [reply nvflare/app_common/ccwf/server_ctl.py:348]: I see. To fix this, you could modify the _update_client_status() method, instead of here, since we want one method to make the right the decision (not spreading decision making to multiple places).
Instead of changing log level to debug, we should change the order of condition checking - check workflow_id first.

**Review by yanchengnv — APPROVED** [review]
LGTM
